### PR TITLE
Feature terraform

### PR DESCRIPTION
**Pull request by jdddog**
This pull request includes:
* The Terraform configuration files for deploying the observatory platform to Google Cloud.
   * The local development environment and the cloud environment are both built with the same Docker and Docker Compose files.
* Refactored DAGs:
  * Making them all as consistent as possible and so that they work with the updated observatory platform:
    * All DAGs check that variables and connections exist as the first step.
    * All DAGs get configuration settings from airflow variables (e.g. project id etc).
    * All DAGs upload downloaded data to Google Cloud storage download bucket.
    * All DAGs cleanup files from local system once finished.
    * All DAGs using @weekly interval, except for Crossref Metadata (the 7th of each month) and Geonames (monthly).
    * Skip operator is used instead of branching operator because it is easier to see what tasks have been skipped on UI.
    * Always choose which files to fetch from a data source based on the last modified or publication date time to make sure that we don't miss releases. I just noticed that we also need to update Fundref to do this as well, that can be another pull request.
   * Crossref metadata: 
       * Extract archive into multiple files and then transform them in parallel to speed up processing. Reduces transform time from 6 hours to 1 hour.
       * Update schema to support has_format field.
   * Fundef:
      * Compress transformed files with gzip.
   * MAG:
      * AuthorExtendedAttributes table schema added. 
* Optimised imports across the project removing unnecessary imports.
* Removed all dependencies in requirements.txt that are not used and updated version numbers where possible.
* Removed all test data files from git and added them to git lfs. The historical files will need to be purged from the git commit history at a later date.
* Removed code that is not being used at this time, it is still in the git history so we still have a copy when we are ready to build those features into the platform again (e.g. OAI-PMH harvesting).
* A few changes to documentation, but I will make another pull request today that includes a refactored README and instructions about installing and using the system including how to deploy the system with Terraform.

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -338,5 +338,44 @@ make html
...
+```bash
+sudo curl -L "https://releases.hashicorp.com/packer/1.6.0/packer_1.6.0_linux_amd64.zip" -o /usr/local/bin/packer
+sudo chmod +x /usr/local/bin/packer
+```
+
+Install Terraform:
+```bash
+sudo curl -L "https://releases.hashicorp.com/terraform/0.13.0-beta3/terraform_0.13.0-beta3_linux_amd64.zip" -o /usr/local/bin/terraform
+sudo chmod +x /usr/local/bin/terraform
+```
```
[the remark this answers is not in the document]
Comment (jdddog) [reply CONTRIBUTING.md:352]: Yeah that would be good. 

There are some instructions for installing the Observatory Platform on Mac OS: https://observatory-platform.readthedocs.io/en/latest/installation.html#system-dependencies-macos.

The instructions for installing Packer and Terraform are now in a tutorial on Read the Docs: https://observatory-platform.readthedocs.io/en/latest/deploy_terraform.html#linux. We should also add the installation instructions for Mac OS as a section here.

Would someone with Mac OS mind testing these instructions and adding the ones for Packer and Terraform?

File: Dockerfile.observatory [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+ARG HOST_USER_ID
+ARG HOST_GROUP_ID
+ARG AO_HOME=/opt/observatory
+
+# Install git which is required when installing dependencies with pip
+USER root
+RUN apt-get update -yqq && apt-get install git build-essential pigz mawk -y
+
+# Change airflow user's user id to the hosts user id
+RUN usermod -u ${HOST_USER_ID} airflow
```
[the remark this answers is not in the document]
Comment (jdddog) [reply Dockerfile.observatory:12]: Yeah we should definitely fix this. Would you be able to open an issue about it?

File: CONTRIBUTING.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (aroelo) [reply CONTRIBUTING.md:352]: Created a PR: https://github.com/The-Academic-Observatory/observatory-platform/pull/223

File: Dockerfile.observatory [hunk 3]
```diff
@@ -0,0 +1,32 @@
+FROM apache/airflow:1.10.11-python3.7
+
+ARG HOST_USER_ID
+ARG HOST_GROUP_ID
+ARG AO_HOME=/opt/observatory
+
+# Install git which is required when installing dependencies with pip
+USER root
+RUN apt-get update -yqq && apt-get install git build-essential pigz mawk -y
```
[the remark this answers is not in the document]
Comment (aroelo) [reply Dockerfile.observatory:9]: See https://github.com/The-Academic-Observatory/observatory-platform/issues/225

File: Dockerfile.observatory (hunk 2 above)
[the remark this answers is not in the document]
Comment (aroelo) [reply Dockerfile.observatory:12]: See https://github.com/The-Academic-Observatory/observatory-platform/issues/226

**Pull request merged**